connector_spec/connector_spec_without_comments.ts

```diff
@@ -0,0 +1,300 @@
+    supportedFeatureIds: Array<"alerting" | "cases" | "uptime" | "security" | "siem" | "generativeAIForSecurity" | "generativeAIForObservability" | "generativeAIForSearchPlayground" | "endpointSecurity">;
+    subFeature?: "endpointSecurity";
+    getKibanaPrivileges?: (args?: {
+        source?: string;
+        params?: Record<string, unknown>;
+    }) => string[];
+}
+
+export const AuthSchema = z.discriminatedUnion("method", [
+    z.object({
+        method: z.literal("none"),
+    }),
+    z.object({
+        method: z.literal("basic"),
+        credentials: z.object({
+            username: z.string(),
+            password: z.string(),
+        }),
+    }),
+    z.object({
+        method: z.literal("bearer"),
+        token: z.string().describe("Bearer token or API key"),
+    }),
+    z.object({
+        method: z.literal("headers"),
+        headers: z.record(z.string(), z.string()),
+    }),
+    z.object({
+        method: z.literal("oauth2"),
+        config: z.object({
+            tokenUrl: z.string().url(),
+            clientId: z.string(),
+            clientSecret: z.string(),
+            scope: z.string().optional(),
+            additionalFields: z.string().optional(),
+        }),
+    }),
+    z.object({
+        method: z.literal("ssl"),
+        certificate: z.discriminatedUnion("type", [
+            z.object({
+                type: z.literal("crt"),
+                cert: z.string().describe("Base64-encoded certificate"),
+                key: z.string().describe("Base64-encoded private key"),
+                passphrase: z.string().optional(),
+                ca: z.string().optional().describe("Base64-encoded CA certificate"),
+                verificationMode: z.enum(["none", "certificate", "full"]).optional(),
+            }),
+            z.object({
+                type: z.literal("pfx"),
+                pfx: z.string().describe("Base64-encoded PFX bundle"),
+                passphrase: z.string().optional(),
+                ca: z.string().optional().describe("Base64-encoded CA certificate"),
+                verificationMode: z.enum(["none", "certificate", "full"]).optional(),
+            }),
+        ]),
+    }),
+    z.object({
+        method: z.literal("aws_sig_v4"),
+        credentials: z.object({
+            accessKey: z.string(),
+            secretKey: z.string(),
+            region: z.string().optional(),
+            service: z.string().optional(),
+        }),
+    }),
+    z.object({
+        method: z.literal("webhook"),
+        url: z.string().url(),
+        extraHeaders: z.record(z.string(), z.string()).optional(),
+    }),
+]);
+
+export type AuthConfig = z.infer<typeof AuthSchema>;
+
+export interface RateLimitPolicy {
+    strategy: "header" | "status_code" | "response_body";
+    codes?: number[];
+    remainingHeader?: string;
+    resetHeader?: string;
+    bodyPath?: string;
+}
+
+export interface PaginationPolicy {
```

Comment:
depending on the API, pagination parameters may be in headers, query parameters, or the request body, I would add also something like - 


```suggestion
export interface PaginationPolicy {
  parameterLocation: "query_params", "headers", "body"
```